types/time.go

```diff
@@ -1468,8 +1468,9 @@ func checkDateRange(t MysqlTime) error {
 }
 
 func checkMonthDay(year, month, day int, allowInvalidDate bool) error {
+	dateTimeInStr := fmt.Sprintf("%d-%d-%d", year, month, day)
```

Comment:
We do not need this line if there's no error, you can put `fmt.Sprintf` directly into `GenWithStackByArgs`.

Reply:
Nice point. Done.